code/modules/power/gravitygenerator.dm

```diff
@@ -290,8 +252,13 @@ GLOBAL_LIST_EMPTY(gravity_generators)
 
 /obj/machinery/gravity_generator/main/update_icon()
 	..()
-	for(var/obj/O in parts)
-		O.update_icon()
+	cut_overlays()
+	if(get_status() != "fix0" && get_status() != "fix1")
+		overlays += "generator_part"
+	if(get_status() == "off")
+		return
+	overlays += "[get_status()]" 
+	overlays += "[overlay_state]"
```

Comment:
We use the procs for adding and removing overlays because it does some behind the scenes stuff with SSoverlays.
```suggestion
	if(get_status() != "fix0" && get_status() != "fix1")
		add_overlay("generator_part")
	if(get_status() == "off")
		return
	add_overlay("[get_status()]")
	add_overlay("[overlay_state]")
```